packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -342,14 +382,35 @@ class HotRunner extends ResidentRunner {
             packageConfig: debuggingOptions.buildInfo.packageConfig,
             projectRootPath: FlutterProject.current().directory.absolute.path,
             fs: globals.fs,
-          ).then((CompilerOutput output) => output?.errorCount == 0)
+          ).then((CompilerOutput output) {
+            compileTimer.stop();
+            totalCompileTime += compileTimer.elapsed;
+            return output?.errorCount == 0;
+          })
         );
       }
+
+      final Stopwatch launchAppTimer = Stopwatch()..start();
       startupTasks.add(device.runHot(
         hotRunner: this,
         route: route,
-      ).then((int result) => result == 0));
+      ).then((int result) {
+        totalLaunchAppTime += launchAppTimer.elapsed;
+        return result == 0;
+      }));
     }
+
+    unawaited(appStartedCompleter?.future?.then((_) => HotEvent('reload-ready',
```

Comment:
this is a new event right?

Reply:
That's right. It's to track the time it spent to launch the app and get hot reload to be ready.

Reply:
instead of attaching it to this future, why not send it in where appStartedCompleter.complete is called? That would be at right about the same time

Reply:
The only reason I do it here is because all the `Stopwatch`es  that collect the metrics are in this method, and `appStartedCompleter.complete` is called from another method. E.g. the `appStartedTimer`, is started when this method starts, and ended when `appStartedCompleter.complete` is called.

Or do you think it's better if I pass the `Stopwatch`, and two existing `Duration`s into the `attach()` method instead?

Reply:
no, I think changing the signature of attach is much worse. this does feel awkward though.